Meta/Lagom/Fuzzers/FuzzTar.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022, Idan Horowitz <idan.horowitz@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/MemoryStream.h>
+#include <LibArchive/TarStream.h>
+#include <stdio.h>
+
+extern "C" int LLVMFuzzerTestOneInput(uint8_t const* data, size_t size)
+{
+    InputMemoryStream input_stream(ReadonlyBytes { data, size });
+    Archive::TarInputStream tar_stream(input_stream);
+
+    if (!tar_stream.valid())
+        return -1;
+
+    for (; !tar_stream.finished(); tar_stream.advance()) {
+        auto const& header = tar_stream.header();
+
+        if (!header.content_is_like_extended_header())
+            continue;
+
+        switch (header.type_flag()) {
+        case Archive::TarFileType::GlobalExtendedHeader: {
+            auto result = tar_stream.for_each_extended_header([&](StringView, StringView) {});
+            if (result.is_error())
+                return -2;
```

Comment:
The [LibFuzzer documentation page](https://llvm.org/docs/LibFuzzer.html) mentions the following:

> // Values other than 0 and -1 are reserved for future use.

> If the fuzz target returns -1 on a given input, libFuzzer will not add that input top the corpus, regardless of what coverage it triggers.

Is that something we'd want?

Reply:
Ah, I copied it from the other fuzzers which are already returning a variety of codes. I'll change mine to 0's, but we might want to open another PR to change all the other fuzzers.